src/main/java/de/tum/cit/aet/artemis/lecture/repository/LectureRepository.java

```diff
@@ -35,6 +35,15 @@ public interface LectureRepository extends ArtemisJpaRepository<Lecture, Long> {
             """)
     Set<Lecture> findAllByCourseId(@Param("courseId") Long courseId);
 
+    @Query("""
+                    SELECT lecture
+                    FROM Lecture lecture
+                    LEFT JOIN FETCH lecture.lectureUnits
+                    WHERE lecture.course.id = :courseId
+                        AND (lecture.visibleDate IS NULL OR lecture.visibleDate <= :now)
+            """)
```

Comment:
This is not aligned with out usual formatting, the query itself looks good
```suggestion
    @Query("""
            SELECT lecture
            FROM Lecture lecture
            LEFT JOIN FETCH lecture.lectureUnits
            WHERE lecture.course.id = :courseId
                AND (lecture.visibleDate IS NULL OR lecture.visibleDate <= :now)
            """)
```